### PR TITLE
Prepare repository for next `.github` release and SBT build improvements

### PR DESCRIPTION
**Pull request by juanpedromoreno**
# What has been done in this PR?

## Prepare project for next `.github` release

- Add `ci-publish` command alias that will be used instead of `ci-release` for publishing artifacts.
- Move docs templates to `docs` folder (as mdoc is expected).
- Move microsite files to `microsite` folder.
- Run `github` before `ci-docs` & `ci-publish`.
- Remove `ci-microsite` (it gets inline in `ci-docs`).
- Add `actions.yml` file that will be used by next `.github` release workflows to read pre/post conditions to run before/after `ci-*` commands.

## Simplify SBT build

- Re-order settings.
- Remove unneeded settings.
- Remove old `sbt-org-policies` markers.
- Add [sbt-remove-test-from-pom](https://github.com/alejandrohdezma/sbt-remove-test-from-pom) to avoid publishing test dependencies to POM.
- Add [sbt-codecov](https://github.com/alejandrohdezma/sbt-codecov) to run tests with coverage and auto-upload to Codecov.

## Update versions

- Update plugins to latest versions.
- Update some dependency versions.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,26 +1,24 @@
 ThisBuild / organization := "io.higherkindness"
-ThisBuild / githubOrganization := "47degrees"
-ThisBuild / crossScalaVersions := Seq("2.12.11", "2.13.1")
+ThisBuild / githubOrganization := "47deg"
```
Comment (alejandrohdezma) [inline build.sbt:2]: Why setting organization to `47deg` (instead of `47degrees`)?

File: build.sbt [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
   .in(file("."))
   .settings(commonSettings)
   .settings(moduleName := "skeuomorph")
 
-lazy val docs = project
-  .in(file("docs"))
+lazy val microsite = project
   .dependsOn(skeuomorph)
-  .settings(moduleName := "skeuomorph-docs")
+  .settings(moduleName := "skeuomorph-microsite")
```
Comment (alejandrohdezma) [inline build.sbt:19]: Why setting a `moduleName` for this module since it isn't being published?

File: build.sbt (hunk 1 above)
Comment (juanpedromoreno) [reply build.sbt]: Both should be fine, I'm restoring 47degrees to minimize changes :)

**Comment by juanpedromoreno** [comment]
@alejandrohdezma ready for another look. Thanks!

**Pull request merged**